velox/expression/tests/ExpressionFuzzer.h

```diff
@@ -48,6 +49,43 @@ class ExpressionFuzzer {
   core::TypedExprPtr generateExpression(const TypePtr& returnType);
 
  private:
+  struct ExprUsageStats {
+    // Num of times the expression was randomly selected.
+    int numTimesSelected = 0;
+    // Num of rows processed by the expression.
+    int numProcessedRows = 0;
+  };
+
+  // A utility class used to keep track of stats relevant to the fuzzer.
+  class ExprStatsListener : public exec::ExprSetListener {
+   public:
+    ExprStatsListener(
+        std::unordered_map<std::string, ExprUsageStats>& exprNameToStats)
+        : exprNameToStats_(exprNameToStats) {}
+
+    void onCompletion(
+        const std::string& uuid,
+        const exec::ExprSetCompletionEvent& event) override {
+      for (auto& [funcName, stats] : event.stats) {
+        auto itr = exprNameToStats_.find(funcName);
+        if (itr == exprNameToStats_.end()) {
+          // Skip expressions like FieldReference and ConstantExpr
+          continue;
+        }
+        itr->second.numProcessedRows += stats.numProcessedRows;
+      }
+    }
+
+    // A no-op since we cannot tie errors directly to functions where they
+    // occurred.
+    void onError(
+        const SelectivityVector& rows,
+        const ::facebook::velox::exec::EvalCtx::ErrorVector& errors) override {}
```

Comment:
We could consider recording the number of error rows for each function in Expr::applyFunction() in the future too.